source/slang/slang-ir-peephole.cpp

```diff
@@ -0,0 +1,110 @@
+#include "slang-ir-peephole.h"
+#include "slang-ir-inst-pass-base.h"
+
+namespace Slang
+{
+struct PeepholeContext : InstPassBase
+{
+    PeepholeContext(IRModule* inModule)
+        : InstPassBase(inModule)
+    {}
+
+    bool changed = false;
+
+    void processInst(IRInst* inst)
+    {
+        switch (inst->getOp())
+        {
+        case kIROp_GetResultError:
+            if (inst->getOperand(0)->getOp() == kIROp_MakeResultError)
+            {
+                inst->replaceUsesWith(inst->getOperand(0)->getOperand(0));
+                changed = true;
```

Comment:
This case doesn't have removeAndDeallocate - is that intended?

Reply:
Thanks for the catch. Will fix.